app/services/trainees/filter.rb

```diff
@@ -68,6 +68,12 @@ def text_search(trainees, text_search)
       trainees.with_name_trainee_id_or_trn_like(text_search)
     end
 
+    def provider(trainees, provider)
+      return trainees if provider.blank?
+
+      trainees.where(provider: Provider.find(provider))
```

Comment:
This could raise a 500 if someone manually enters an incorrect provider id, should we perhaps use `.find_by` or `where(provider_id: provider)`. Conscious that this will also affect https://github.com/DFE-Digital/register-trainee-teachers/pull/1710/files#diff-915715d62923f715f031d152f01cfd1b70f47b07606e89127e2fe13611798861R9

Reply:
Thanks! Update in 6e6eb6a.